### ISSUE TITLE
Change item creation form

### ISSUE THREAD

**Issue by mkrzmr** [SSHOC/sshoc-marketplace-frontend#180]
related to #179

With the change in recommended/optional metadata fields, change the item creation form accordingly so that only recommended fields are selected. 

Move "Version", which is hardly used and optional, down to the other properties

For all item types except Workflows, make accessibleAt mandatory to complete the form and submit the entry

**Comment by laureD19** [comment]
+ 1 for the necessity to update the edit forms based on the new list of recommended metadata fields, including "downgrading" the version field in the edit forms.

However, for the latest point mentioned above - "make accessibleAt mandatory" - this was extensively discussed during the  development phase of the MP and would require changes to the data model as well as in backend. With our current metadata population, it would also create a lot of bugs as we still have a bunch of items without "accessibleAt" but already in the system. So, even if this point has been spotted as an important one for future upgrade of the data model, I don't think we should implement it now. I would recommend instead to list it as a potential new requirement for the next MP upgrade.

**Comment by laureD19** [comment]
after discussion, agreement on making `accessibleAt` mandatory.
Still, the rest of the recommended fields + downgrading `version` should be updated in the edit forms.

**Comment by babslgam** [comment]
> For all item types except Workflows, make accessibleAt mandatory to complete the form and submit the entry

@stefanprobst is it even already possible, to add a mandatory field based on item type? 
because isRequired would be defined here, not?
https://github.com/SSHOC/sshoc-marketplace-frontend/blob/main/src/components/item-form/useItemFormFields.ts#L50-L55

**Comment by stefanprobst** [comment]
yes, IIRC you should be able to mark `accessibleAt` with `isRequired: true` in `useItemFormFields`, and overwrite with `isRequired: false` in [`useWorkflowFormFields`](https://github.com/SSHOC/sshoc-marketplace-frontend/blob/main/src/components/item-form/useWorkflowFormFields.ts) and [`useWorkflowStepFormFields`](https://github.com/SSHOC/sshoc-marketplace-frontend/blob/main/src/components/item-form/useWorkflowStepFormFields.ts)

---

> Move "Version", which is hardly used and optional, down to the other properties

@mkrzmr `version` is not a property in the data model, but an optional top-level field, so we cannot really integrate it with other properties.

**Comment by babslgam** [comment]
> [@mkrzmr](https://github.com/mkrzmr) `version` is not a property in the data model, but an optional top-level field, so we cannot really integrate it with other property fields.

whats https://github.com/SSHOC/sshoc-marketplace-frontend/blob/main/src/dictionaries/authenticated/index.ts#L160 then?

**Comment by stefanprobst** [comment]
> Can we change where it appears on the page? It seems quite important because of where it is, but is hardly used and throws people off when they don't know if that means version of software or version of entry.

since `version` is a top-level metadata field, it is currently displayed together with other top-level metadata fields. we can of course display it somewhere else, but we definitely don't want to create a separate section for it. so where should it be moved?

maybe this is also an indicator that `version` should be demoted to just another `properties` field - but that would be a breaking change to the data model and would require changes to backend, api versioning, and content migration.

alternatively, if the main issue is confusion *what* version this field represents - can we just add a help text underneath it (like other fields already have)?

**Comment by stefanprobst** [comment]
> whats https://github.com/SSHOC/sshoc-marketplace-frontend/blob/main/src/dictionaries/authenticated/index.ts#L160 then?

the "version" textfield in the initial section of item create/edit forms populates this `version` field:

https://github.com/SSHOC/sshoc-marketplace-frontend/blob/4e196db1d9eab2eb281a40233bb801e61df06cff/src/data/sshoc/api/item.ts#L217

see also e.g. the `ToolCore` schema in [openapi](https://marketplace-api.sshopencloud.eu/swagger-ui/index.html#/model-ToolCore) docs.

the `properties` field *also* allows a `version` string property:

```
curl https://marketplace-api.sshopencloud.eu/api/property-types/version | jq
```

but properties are displayed/edited in their own form section already:

![Image](https://github.com/user-attachments/assets/660dcf2b-027c-48ad-83f1-c2d1fd0482bd)

(specifically, the `Dictionary` type you linked to refers to the type of i18n messages)

**Comment by laureD19** [comment]
Thanks for the changes!

I have two questions/comments (maybe more for you @KlausIllmayer ):
1. what does it mean to turn "accessibleAt" in a mandatory field, esp. for already created items that have no value in this field? Is it mandatory only for newly created items? 

2. Regarding the 'version' fieldS, I agree with Stefan > maybe this is also an indicator that version should be demoted to just another properties field - but that would be a breaking change to the data model and would require changes to backend, api versioning, and content migration.

I had a quick look and we currently have 233 items with value in the top-level metadata field 'version' that we could move to the dynamic property 'version', before asking changes to backend. => [issue for curation team](https://github.com/SSHOC/marketplace-curation/issues/39). Any contraindications to this approach @vronk @mkrzmr @KlausIllmayer ?

**Comment by KlausIllmayer** [comment]
I guess by turning `accessibleAt` in a mandatory field we are talking about doing this only in frontend and not in backend, true? If so, then it is in my understanding only a convention, as for backend it does not matter if there is a value in accessibleAt or not. It could be, that frontend has its troubles when you call an item without an `accessibleAt`. That must be then covered in the frontend code. I remember that we had a long discussion about the mandatory character of `accessibleAt`. I argued, that it can be a kind of history record, that there was once a tool that was widely used but as the website shut down, it is not available any more online. But this will not happen often and it can be argued, that marketplace is not the place for keeping historical data (anyways, chances are high that in such case there is a link to the once existing website on the Wayback machine of the Internet Archive (https://archive.org/) that can be used for `accessibleAt`. The only issue that still comes into my mind is that workflows do not have `accessibleAt`-values, so this must be kept in mind. And it should be also discussed, if backend is the better place to make `accessibleAt` mandatory. In such a case we would need to think how we deal with old items that do not have an `accessibleAt`.

Regarding version: I think it is a good idea to move it to a dynamic property. We also discussed this long ago and if I remember correct, we all agreed but postponed it due to the necessary changes in backend. The workflow would be to create the dynamic property and then to move the current values of the field `version` to the dynamic property (could be done with the help of a script) and then asking backend team to remove the field `version`.

**Comment by stefanprobst** [comment]
ad `accessibleAt`: this has already been implemented by barbara: the frontend validates client-side that `accessibleAt` has at least one entry. the backend does not validate anything currently.

ad `version`: afaiu we currently have *both* a top-level `version` field, and a `version` property (https://marketplace-api.sshopencloud.eu/api/property-types/version) - not sure if they mean the same though (i.e. sshoc metadata version? entity version?)
